src/lib/components/Matrix/Matrix.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import MatrixItem from './MatrixItem';
+
+class Matrix extends React.Component {
+  constructor() {
+    super();
+    this.generateExtras = this.generateExtras.bind(this);
+  }
+
+  generateExtras() {
+    const { children } = this.props;
+    const extraCount = (2 * React.Children.count(children)) % 3;
+    const extraItems = [];
```

Comment:
My only comment is that I didn't immediately realise what extraCount and extraItems related to.
If this could be slightly clearer, I think it would be nicer.

As it is the items to fill in the remaining space on a row. It could possibly be `emptyItemCount` or similar.